src/image.cpp

```diff
@@ -221,31 +227,21 @@ uint16_t Image::byteSwap(uint16_t value, bool bSwap) {
 }
 
 uint16_t Image::byteSwap2(const DataBuf& buf, size_t offset, bool bSwap) {
-  uint16_t v = 0;
-  auto p = reinterpret_cast<char*>(&v);
-  p[0] = buf.read_uint8(offset);
-  p[1] = buf.read_uint8(offset + 1);
+  uint16_t v;
+  std::memcpy(&v, buf.c_data(offset), sizeof(uint16_t));
```

Comment:
The `read_uint8` method checks for out-of-bound accesses, so it isn't a good idea to replace it with a `memcpy`.

It would be better to use methods like [`read_uint16`](https://github.com/Exiv2/exiv2/blob/376e3f9049ae03306e985e199b7257028daf9e6c/include/exiv2/types.hpp#L175) and [`read_uint32`](https://github.com/Exiv2/exiv2/blob/376e3f9049ae03306e985e199b7257028daf9e6c/include/exiv2/types.hpp#L178) which have a `ByteOrder` parameter. But this function has a `bSwap` parameter which doesn't mean the same thing, so a bit of refactoring would be needed.